### PR TITLE
drivers: video: add isx019 camera sensor

### PR DESCRIPTION
**Pull request by jerpelea**
## Summary
ISX019 has the DOL-HDR feature, which combines images with different exposure so that
both bright and dark data of high-contrast subject can be viewed at the same time.

## Impact
Camera driver

## Testing
Spresense

### CODE REVIEW

**Comment by jerpelea** [comment]
please ignore the false nxstyle error

File: drivers/video/isx019.c [hunk 1]
```diff
@@ -0,0 +1,3249 @@
...
+
+static void fpga_unlock(void)
+{
+  nxsem_post(&g_isx019_private.fpga_lock);
+}
+
+static int fpga_i2c_write(uint8_t addr, FAR uint8_t *data, uint8_t size)
+{
+  struct i2c_config_s config;
+  static uint8_t buf[FPGA_I2C_REGSIZE_MAX + FPGA_I2C_REGADDR_LEN];
```
Comment (pkarashchenko) [inline drivers/video/isx019.c:787]: Do we really need `static` here? I mean, that I do not see same approach in other places in this file

**Review by pkarashchenko — APPROVED** [review]
All comments are optional.

File: include/nuttx/video/isx019.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+int isx019_initialize(void);
+int isx019_uninitialize(void);
+#ifdef CONFIG_VIDEO_ISX019_REGDEBUG
+int isx019_read_register(uint8_t  cat,
+                         uint16_t addr,
+                         uint8_t  *buf,
```
Comment (pkarashchenko) [inline include/nuttx/video/isx019.h:47]:
```suggestion
                         FAR uint8_t  *buf,
```

File: drivers/video/isx019_range.h [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __DRIVERS_VIDEO_ISX019_RANGE_H
+#define __DRIVERS_VIDEO_ISX019_RANGE_H
+
```
Comment (pkarashchenko) [inline drivers/video/isx019_range.h:23]: need to add `#include <limits.h>` and `#include <nuttx/video/imgsensor.h>`

File: drivers/video/isx019.c [hunk 4]
```diff
@@ -0,0 +1,3249 @@
...
+
+int isx019_initialize(void)
+{
+  imgsensor_register(&g_isx019_ops);
+  nxsem_init(&g_isx019_private.i2c_lock, 0, 1);
+  nxsem_init(&g_isx019_private.fpga_lock, 0, 1);
+  return OK;
+}
+
+int isx019_uninitialize()
```
Comment (pkarashchenko) [inline drivers/video/isx019.c:3223]:
```suggestion
int isx019_uninitialize(void)
```

File: drivers/video/isx019.c [hunk 5]
```diff
@@ -0,0 +1,3249 @@
...
+  return OK;
+}
+
+#ifdef CONFIG_VIDEO_ISX019_REGDEBUG
+int isx019_read_register(uint8_t cat,
+                         uint16_t addr,
+                         FAR uint8_t *buf,
+                         uint8_t size)
+{
+  int ret = -EINVAL;
```
Comment (pkarashchenko) [inline drivers/video/isx019.c:3236]:
```suggestion
  int ret;
```

File: drivers/video/isx019.c [hunk 6]
```diff
@@ -0,0 +1,3249 @@
...
+          break;
+        }
+    }
+
+  return ret;
+}
+
+static int validate_value(uint32_t id,
+                          uint32_t size,
+                          FAR imgsensor_value_t val)
```
Comment (pkarashchenko) [inline drivers/video/isx019.c:3139]:
```suggestion
                          imgsensor_value_t val)
```

File: drivers/video/isx019.c [hunk 7]
```diff
@@ -0,0 +1,3249 @@
...
+        ret = validate_range(val.value32, &sup.u.range);
+        break;
+    }
+
+  return ret;
+}
+
+static int isx019_set_value(uint32_t id,
+                            uint32_t size,
+                            FAR imgsensor_value_t val)
```
Comment (pkarashchenko) [inline drivers/video/isx019.c:3178]:
```suggestion
                            imgsensor_value_t val)
```

File: drivers/video/isx019.c [hunk 8]
```diff
@@ -0,0 +1,3249 @@
...
+  int32_t jpg_quality;
+  imgsensor_colorfx_t colorfx;
+};
+
+typedef struct isx019_dev_s isx019_dev_t;
+
+typedef CODE int32_t (*convert_t)(int32_t value32);
+
+typedef CODE int (*setvalue_t)(imgsensor_value_t value);
+typedef CODE int (*getvalue_t)(imgsensor_value_t *value);
```
Comment (pkarashchenko) [inline drivers/video/isx019.c:217]:
```suggestion
typedef CODE int (*getvalue_t)(FAR imgsensor_value_t *value);
```

**Comment by jerpelea** [comment]
@pkarashchenko thanks

File: drivers/video/isx019.c [hunk 9]
```diff
@@ -0,0 +1,3249 @@
...
+
+#define TRANSITION_TIME_TO_STARTUP   (120 * USEC_PER_MSEC) /* unit : usec */
+#define TRANSITION_TIME_TO_STREAMING (30 * USEC_PER_MSEC)  /* unit : usec */
+
+/* For get_supported_value() I/F */
+
+#define SET_RANGE(range, min, max, s, def) \
+        do                                 \
+          {                                \
+            (range.minimum        = (min); \
```
Comment (pkarashchenko) [inline drivers/video/isx019.c:57]:
```suggestion
            (range).minimum       = (min); \
```

File: drivers/video/isx019.c (hunk 1 above)
Comment (pkarashchenko) [reply drivers/video/isx019.c:787]: I assume that "resolved" means that `static` is intended to be applied here

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**